src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,42 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.model.RegistrationException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MINIMUM_LENGTH = 6;
+    private static final int MINIMUM_AGE = 18;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user.getPassword() == null) {
+            throw new RegistrationException("Password cannot be null");
+        }
+        if (user.getLogin() == null) {
+            throw new RegistrationException("Login cannot be null");
+        }
+        if (user.getAge() == null) {
+            throw new RegistrationException("Age cannot be null");
+        }
+        if (user.getPassword().length() < MINIMUM_LENGTH) {
+            throw new RegistrationException("Not valid password " + user.getPassword()
+                    + ".Password cannot be less than " + MINIMUM_LENGTH + " characters");
+        }
+        if (user.getLogin().length() < MINIMUM_LENGTH) {
+            throw new RegistrationException("Not valid login " + user.getLogin()
+                    + ".Login cannot be less than " + MINIMUM_LENGTH + " characters");
+        }
+        if (user.getAge() < MINIMUM_AGE) {
+            throw new RegistrationException("Not valid age: " + user.getAge()
+                    + ".Age cannot be less than " + MINIMUM_AGE);
+        }
+        if (storageDao.get(user.getLogin()) == null) {
+            storageDao.add(user);
+            return user;
+        } else {
+            throw new RegistrationException("A user with this login already exists");
```

Comment:
The check for an existing user by login should be done before any other validations to avoid unnecessary operations in case the user already exists. This way, you don't perform validations for a user that won't be registered anyway.